docs/source/app-dev/services.rst

```diff
@@ -0,0 +1,159 @@
+
+Reading from the ledger
+***********************
+
+Transaction service
+===================
+
+Use the **transaction service** to listen to changes in the ledger state, reported via a stream of transaction events.
+
+Transaction events detail the changes on transaction boundaries - each event denotes a transaction on the ledger, and contains all the update events (create, exercise, archive of contracts) that had an effect in that transaction.
+
+Transaction events contain a :ref:`transactionId <com.digitalasset.ledger.api.v1.Transaction.transaction_id>` (assigned by the server), the ``workflowId``, the ``commandId``, and the events in the transaction.
+
+Transaction events are the primary mechanism by which an application will do its work. Event-driven applications can use them to generate new commands, and state-driven applications will use them to update their state model, by e.g. creating data that represents created contracts.
+
+The transaction service can be initiated to read events from an arbitrary point on the ledger. This is important when starting or restarting and application, and works in conjunction with the `active contract service <#active-contract-service>`__.
+
+Transaction and transaction trees
+---------------------------------
+
+``TransactionService`` offers several different subscriptions. The most commonly used is ``GetTransactions``. If you need more details, you can use ``GetTransactionTrees`` instead, which returns transactions as flattened trees, represented as a map of event IDs to events and a list of root event IDs.
+
+.. _verbosity:
+
+Verbosity
+---------
+
+The service works in a non-verbose mode by default, which means that some identifiers are omitted:
+
+- Record IDs
+- Record field labels
+- Variant IDs
+
+You can get these included in requests related to Transactions by setting the ``verbose`` field in message ``GetTransactionsRequest`` or ``GetActiveContractsRequest`` to ``true``.
+
+Active contract service
+=======================
+
+Use the **active contract service** to obtain a party-specific view of all the contracts recently active on the ledger.
+
+The active contract service returns the current contract set as a set of created events that would re-create the state being reported, along with the ledger position at which the view of the set was taken.
+
+For state-driven applications, this is most important at application start. They must synchronize their initial state with a known view of the ledger, and without this service, the only way to do this would be to read the Transaction Stream from the beginning of the ledger. This can be prohibitive with a large ledger.
```

Comment:
```suggestion
This is most important at application start, if the application needs to synchronize its initial state with a known view of the ledger. Without this service, the only way to do this would be to read the Transaction Stream from the beginning of the ledger, which can be prohibitively expensive with a large ledger.
```